### PR TITLE
Port Muon ALC Interface

### LINKED ISSUE #23476: Make ALC interface Mantid 4 compatible
The muon ALC interface should be made Mantid 4 compatible (i.e. replace plotting)

### PR DESCRIPTION
**Pull request by robertapplin**
**Description of work.**
This PR ports the Muon ALC interface.

It creates the basics for a RangeSelector which is useable on mantid workbench, and it also creates a PeakPicker tool.

**To test:**
Test on MantidPlot and Workbench.
**Part 1**
1. Set the default instrument to `EMU`, and turn on the data archive
2.`Interfaces`->`Muon`->`ALC`
3. For the first data put `81061` and for last put `81142`
4. Select `field_danfysik` for the log value
5. Tick the subtract box in the periods section and set the second combo box to 2.
6. In Calculation, set the max to 16
7. Click run and wait for the data to load

**Part 2**
1. Click `Baseline Modelling`
2. Add a `linear background` fit function
3. Add 2 fitting regions (by right clicking and adding sections)
4. Set one to be before the peak and one to be after the peak
5. Do the fit

**Part 3**
1. Go to the `Curve Fitting`
2. Add a Gaussian (put sigma = 20, centre = 2150 and height =0.02)
3. Select the Grey line by clicking on it. This will make the peak picker active which shows if you have a good sigma.
3. Do a fit

Fixes #23476

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/python/mantidqt/widgets/fitpropertybrowser/markers.py [hunk 1]
```diff
@@ -88,12 +112,26 @@ def is_above(self, x, y):
         :return: True or False.
         """
         x_pixels, y_pixels = self.patch.get_transform().transform((x, y))
-        if self.y0 is not None and y < self.y0:
-            return False
-        if self.y1 is not None and y > self.y1:
-            return False
+        if self.check_y_height:
```
Comment (AnthonyLim23) [inline qt/python/mantidqt/widgets/fitpropertybrowser/markers.py]: you could add this to the other if statements with an `and`

File: qt/python/mantidqt/widgets/fitpropertybrowser/markers.py [hunk 2]
```diff
@@ -136,14 +180,18 @@ def override_cursor(self, x, y):
...
+            if self.y0 is not None and y < self.y0:
+                return None
+            if self.y1 is not None and y > self.y1:
+                return None
         if self.is_moving or self.is_above(x, y):
             return self.get_cursor_at_y(y)
         return None
 
+    def check_y_height(self, check):
```
Comment (AnthonyLim23) [inline qt/python/mantidqt/widgets/fitpropertybrowser/markers.py]: you should use setters and getters geeksforgeeks

**Comment by AnthonyLim23** [comment]
The python interfaces have vanished in workbench. @martyngigg do you have any idea why this might have happened?

**Comment by AnthonyLim23** [comment]
If I try to change the locations of the fitting lines by directly entering the number into the GUI, the lines do not update. It does update after adding/removing a section. I think its just a missing signal

If you make the GUI full screen and then change to the next stage, the bottom of the GUI will go off screen.

**Comment by martyngigg** [comment]
@AnthonyLim23 That's odd. I can see them myself if I build the branch. If you run `print(config['mantidqt.python_interfaces_directory'])` what does it say?

**Comment by robertapplin** [comment]
@AnthonyLim23 I cannot reproduce your second comment, can you show me?

**Comment by AnthonyLim23** [comment]
I got 

```
C:/Users/BTR75544/work/dev/mantid/scripts/
Loading libraries from 'C:/Users/BTR75544/work/dev/Build/bin/Debug/', excluding 'Qt4;Qt5'
Opening all libraries in C:/Users/BTR75544/work/dev/Build/bin/Debug/
```

The GUI's did not reappear afterwards.

**Comment by martyngigg** [comment]
That's odd. I can't see any changes in this PR that would affect the loading of the Python GUIs in any way. 

What does 
```
print(config['mantidqt.python_interfaces'])
```
say?
Is this key in your `Mantid.user.properties`?

**Comment by AnthonyLim23** [comment]
I got
```
Direct/DGS_Reduction.py Direct/DGSPlanner.py Direct/PyChop.py Direct/MSlice.py SANS/ORNL_SANS.py Utility/TofConverter.py Reflectometry/ISIS_Reflectometry_Old.py Diffraction/Powder_Diffraction_Reduction.py Utility/FilterEvents.py Diffraction/HFIR_4Circle_Reduction.py Utility/QECoverage.py SANS/ISIS_SANS.py Muon/Frequency_Domain_Analysis.py Muon/Elemental_Analysis.py Muon/Frequency_Domain_Analysis_Old.py Muon/Muon_Analysis_2.py
```

and my Mantid.user.properties is:

```
# This file can be used to override any properties for this installation.
# Any properties found in this file will override any that are found in the Mantid.Properties file
# As this file will not be replaced with further installations of Mantid it is a safe place to put 
# properties that suit your particular installation.
#
# See here for a list of possible options:
# http://docs.mantidproject.org/nightly/concepts/PropertiesFile.html

##
## GENERAL
##

## Set the number of algorithm properties to retain
#algorithms.retained=90

## Set the maximum number of cores used to run algorithms over
#MultiThreaded.MaxCores=4

##
## FACILITY AND INSTRUMENT
##

## Sets the default facility
## e.g.: ISIS, SNS, ILL
default.facility=ISIS

## Sets the default instrument
## e.g. IRIS, HET, NIMROD
default.instrument=EMU


## Sets the Q.convention
## Set to Crystallography for kf-ki instead of default Inelastic which is ki-kf
#Q.convention=Crystallography
##
## DIRECTORIES
##

## Sets a list of directories (separated by semi colons) to search for data
#datasearch.directories=../data;../isis/data

## Set a list (separated by semi colons) of directories to look for additional Python scripts
#pythonscripts.directories=../scripts;../docs/MyScripts

## Uncomment to enable archive search - ICat and Orbiter
#datasearch.searcharchive=On

## Sets default save directory
#defaultsave.directory=../data

##
## LOGGING
##

## Uncomment to change logging level
## Default is information
## Valid values are: error, warning, notice, information, debug
#logging.loggers.root.level=information

##
## MantidPlot
##

## Hides categories from the algorithm list in MantidPlot
#algorithms.catagories.hidden=Muons,Inelastic

## Show invisible workspaces
#MantidOptions.InvisibleWorkspaces=0
## Re-use plot instances for different plot types
#MantidOptions.ReusePlotInstances=Off

## Uncomment to disable use of OpenGL to render unwrapped instrument views
#MantidOptions.InstrumentView.UseOpenGL=Off

datasearch.directories=C:/Users/BTR75544/work/dev/mantid/instrument/;C:/Users/BTR75544/work/shared/;C:/Users/BTR75544/work/shared/Sundials/

datasearch.searcharchive=all
```

**Comment by martyngigg** [comment]
This is weird. Can you add some print statements to [_discover_python_interfaces](https://github.com/mantidproject/mantid/blob/master/qt/applications/workbench/workbench/app/mainwindow.py#L379) to see what is happening?

**Comment by AnthonyLim23** [comment]
at the end I had:
```
 {u'Muon': [u'Frequency_Domain_Analysis.py'], u'Utility': [u'TofConverter.py', u'FilterEvents.py', u'QECoverage.py'], u'Diffraction': [u'Powder_Diffraction_Reduction.py', u'HFIR_4Circle_Reduction.py'], u'Direct': [u'DGS_Reduction.py', u'DGSPlanner.py', u'PyChop.py', u'MSlice.py'], u'SANS': [u'ORNL_SANS.py', u'ISIS_SANS.py']}
```
which is missing muon analysis 2. But FDA does not show up in the muon list either.

**Comment by AnthonyLim23** [comment]
I'm happy with it working in workbench. I have fixed the interface menu bug too. @martyngigg could you review the code please?

**Review by martyngigg — CHANGES_REQUESTED** [review]
This is the correct fix. I can now see both C++ and Python interfaces for a given category.

A minor code fix to follow the coding style.

File: qt/applications/workbench/workbench/app/mainwindow.py [hunk 3]
```diff
@@ -395,19 +395,21 @@ def _discover_python_interfaces(self, interface_dir):
 
         return interfaces
 
-    def _discover_cpp_interfaces(self):
+    def _discover_cpp_interfaces(self, Interfaces):
```
Comment (martyngigg) [inline qt/applications/workbench/workbench/app/mainwindow.py]: Can we following the coding conventions and surrounding code and lower-case the `I`? Can we also update the method docstring to include a description of the argument?

**Review by martyngigg — APPROVED** [review]

**Pull request merged**